web/controllers/package_controller.ex

```diff
@@ -3,24 +3,43 @@ defmodule HexWeb.PackageController do
 
   @packages_per_page 30
   @sort_params ~w(name downloads inserted_at)
+  @letters for letter <- ?A..?Z, do: <<letter>>
 
   def index(conn, params) do
+    letter        = HexWeb.Utils.safe_search(params["letter"])
     search        = HexWeb.Utils.safe_search(params["search"])
+    maintainer    = params["maintainer"]
+
+    filter =
+      cond do
+        letter ->
+          {:starts_with, letter}
+        maintainer ->
+          {:maintainer, maintainer}
```

Comment:
We shouldn't add maintainer search at this time, that information is currently private.